api/v1beta1/verticadb_types.go

```diff
@@ -555,6 +563,35 @@ type CommunalStorage struct {
 	// Name of the Kerberos realm.  This is set in the database config parameter
 	// KerberosRealm during bootstrapping.
 	KerberosRealm string `json:"kerberosRealm,omitempty"`
+
+	// +kubebuilder:default:=""
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:select:SSE-S3","urn:alm:descriptor:com.tectonic.ui:select:SSE-KMS","urn:alm:descriptor:com.tectonic.ui:select:SSE-C"}
+	// The server-side encryption type Vertica will use to read/write from encrypted S3 communal storage.
+	// Available values are: SSE-S3, SSE-KMS, SSE-C and empty string ("").
+	// - SSE-S3: the S3 service manages encryption keys.
+	// - SSE-KMS: encryption keys are managed by the Key Management Service (KMS).
+	// 	 KMS key identifier must be supplied through communal.additionalConfig map.
+	// - SSE-C: the client manages encryption keys and provides them to S3 for each operation.
+	// 	 The client key must be supplied through communal.s3SseCustomerKeySecret.
+	// - Empty string (""): No encryption. This is the default value.
+	// This value cannot change after the initial creation of the VerticaDB.
+	S3ServerSideEncryption ServerSideEncryptionType `json:"s3ServerSideEncryption,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:io.kubernetes:Secret"
+	// The name of a secret that contains the key to use for the S3SseCustomerKey config option in the server.
+	// It is required when S3ServerSideEncryption is SSE-C. When set, the secret must have a key named clientKey.
+	S3SseCustomerKeySecret string `json:"s3SseCustomerKeySecret,omitempty"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// Contains a map of server configuration parameters.
+	// To avoid duplicate values, if a parameter is already set through another CR field,
+	// (like S3ServerSideEncryption through communal.s3ServerSideEncryption), the corresponding
+	// key/value pair is skipped.
+	// These are set only during initial bootstrap. After the database has been initialized,
+	// changing the options in the CR will have no affect in the server.
```

Comment:
Suggestion call out what happens if you use a parameter that isn't supported by your server version:
```suggestion
	// To avoid duplicate values, if a parameter is already set through another CR field,
	// (like S3ServerSideEncryption through communal.s3ServerSideEncryption), the corresponding
	// key/value pair is skipped. If a config value is set that isn't supported by the server version
	// you are running, the server will fail to start. These are set only during initial bootstrap. After
	// the database has been initialized, changing the options in the CR will have no affect in the server.
```